dch/public/app/components/navbar/canary-alerts/canary-alerts-service.js

```diff
@@ -0,0 +1,131 @@
+(function() {
+
+  'use strict';
+
+  /**
+   * @desc - This will be responsible for both receiving alert streams from
+   *         the DCH server and fetching all alerts during initialization.
+   *         It will also be responsible for issuing deletion commands.
+   *
+   * @ngInject
+   *  - $http
+   */
+  angular
+    .module('dcgui.shared')
+    .service('CanaryAlertService', CanaryAlertService);
+
+  CanaryAlertService.$inject = ['$http'];
+
+  function CanaryAlertService($http) {
+    var canaryAlertService = {
+      getWatchAlerts: getWatchAlerts,
+      regiterCallback: regiterCallback,
+      deleteWatchAlert: deleteWatchAlert
+    };
+
+    var ALERT_STREAM_PATH = 'api/alert-sse';
+    var ALERT_EVENT = 'alert-event';
+    var GET_ALERTS_PATH = '/get-alerts';
```

Comment:
Has this been implemented? We should be able to populate the inbox with messages already stored in the DB using this api